config/finders.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/adrg/xdg"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+)
+
+type Finder func(cfg Config) []string
+
+// FindDirect attempts to find a directly configured cfg.File
+func FindDirect(cfg Config) []string {
+	if cfg.File == "" {
+		return nil
+	}
+	file, err := homedir.Expand(cfg.File)
+	if err != nil {
+		cfg.Logger.Debugf("unable to expand path: %s", cfg.File)
+		file = cfg.File
+	}
+	if fileExists(file) {
+		return []string{file}
+	}
+	return nil
+}
+
+// FindInCwdConfigYaml loads ./config.yaml
+// NOTE: THIS IS WAS A BUG IN SYFT/et. al. and is DEPRECATED and SHOULD NOT BE USED
+// it is just here for temporary compatibility -- See syft #1634
+func FindInCwdConfigYaml(cfg Config) []string {
```

Comment:
the reason for breaking these out into function is so that Syft can implement this function for itself and add it to the list of finders during initialization. That means we don't need to implement this here at all and immediately not recommend it's use.

Reply:
Understood -- this isn't a default thing, but it's an option for a user who wants to load `./config.yaml`. We could remove it before anyone uses this library if this is an especially bad idea.